### PR TITLE
DROOLS-5673: [DMN Designer] Validation fails when an included node is added on a DRD

### PR DESCRIPTION
**Pull request by karreiro**
**JIRA**: [DROOLS-5673](https://issues.redhat.com/browse/DROOLS-5673): [DMN Designer] Validation fails when the included node is added on a DRD

**Artifacts**
- **Business Central (reproducible on BC only)**: [WAR file v2](https://www.dropbox.com/s/v79o2zmi6s1r6mo/business-central-DROOLS-5673-2.war?dl=0)
- **VSCode**: [plugin v2](https://www.dropbox.com/s/tfv04mdtri5n73c/plugin-DROOLS-5673-2.vsix?dl=0)

**Before:**
bug

**After:**

---

<pre>
To retest a PR or trigger a specific build please add a comment:

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</pre>

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins do fdb.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/docks/navigator/included/components/DecisionComponents.java [hunk 1]
```diff
@@ -88,41 +97,82 @@ public View getView() {
...
+                createDecisionComponentItem(component);
+            }
         }
     }
 
+    private boolean isDRGElementAdded(final DecisionComponent decisionComponent) {
+        return getDecisionComponentsItems().stream().anyMatch(item -> {
+            final String decisionComponentItemName = item.getDecisionComponent().getName();
+            final String decisionComponentName = decisionComponent.getName();
+            return Objects.equals(decisionComponentItemName, decisionComponentName);
```
[the remark this answers is not in the document]
Comment (jstastny-cz) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/docks/navigator/included/components/DecisionComponents.java:172]: even the namespace can be mimicked - include model with `ddddd` prefix, resulting in `ddddd.dec2` decision. User can duplicate such name and the panel will treat those two as a single component.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/docks/navigator/included/components/DecisionComponents.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/docks/navigator/included/components/DecisionComponents.java]: Good catch, @jstastny-cz. Thank you. Now the checks are based on the DRG element id, to solve the issue.

Also, this is solved for nodes with mimicked namespaces as well, they will appear correctly in the Decision Components dock.

**Comment by karreiro** [comment]
Thank you for the review, @jstastny-cz. Changes applied on a separate commit, and I've updated artifacts in the description.

**Comment by karreiro** [comment]
Jenkins do fdb.

**Review by danielzhe — APPROVED** [review]
Thank you, @karreiro

**Comment by karreiro** [comment]
Jenkins do fdb.

**Comment by karreiro** [comment]
Jenkins retest this.

**Comment by karreiro** [comment]
Jenkins retest this.

**Review by jstastny-cz — APPROVED** [review]
Looks good, verified in BC.

**Pull request merged**